### ISSUE TITLE
Attribution links only identified by colour

### ISSUE THREAD

**Issue by danallsop** [ScottLogic/jumpstart-Leaflet#229]
> **Migrated from:** https://github.com/Leaflet/Leaflet/issues/6742
> **Original author:** danallsop
> **Original created:** 2019-07-30T15:22:59Z
---

The maps have 'leaflet' link which is identified only by color.

Criteria 1.4.1
Compliance success level A

![0f49f0d4-0e1d-11e7-8e12-1a83beefc03c](https://user-images.githubusercontent.com/871728/62141892-52d79780-b2bb-11e9-925d-2046f703f4cb.png)

- [x] I've looked at the [documentation](http://leafletjs.com/reference.html) to make sure the behavior is documented and expected
- [x] I'm sure this is a Leaflet code issue, not an issue with my own code nor with the framework I'm using (Cordova, Ionic, Angular, React…)
- [x] I've searched through the issues to make sure it's not yet reported

**Steps to reproduce**
- View a map
- Look at the link text in the bottom right
- No visual indication that text is a link other than the colour difference

**Expected behavior**
When the link has focus or hover it should show an underline or some other visual indicator to pass WCAG compliance.
See https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-without-color.html

**Current behavior**
Only identified by a colour.
See https://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20161007/F73 for an example failure.

**Environment**
- Leaflet version: all
- Browser (with version): all
- OS/Platform (with version): all

**Comment by danallsop** [comment]
## Comments (2 total)

---
> **Comment by:** danallsop **on** 2021-04-01T14:29:50Z

> See https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-without-color.html

The latest recommendation is https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html.

> No visual indication that text is a link other than the colour difference

According to [Technique G183](https://www.w3.org/WAI/WCAG21/Techniques/general/G183.html) as long as the difference in lightness (contrast) is 3:1 or greater with the surrounding text they still pass <span title="Success Criterion">SC</span> [1.4.1 Use of Color](https://www.w3.org/TR/WCAG21/#use-of-color). That contrast ratio is currently 2.56 (fails SC [1.4.3 Contrast (Minimum)](https://www.w3.org/TR/WCAG21/#contrast-minimum) and [1.4.6 Contrast (Enhanced)](https://www.w3.org/TR/WCAG21/#contrast-enhanced)).

> **Expected behavior**
> When the link has focus or hover it should show an underline

[Technique F73](https://www.w3.org/WAI/WCAG21/Techniques/failures/F73.html) states:

<q>If the non-color cue only happens when the mouse hovers over the link or when the link receives focus, it is still a failure.</q>
so that wouldn't be sufficient, it'd need to always have an underline.

It also states: <q>If the link is a different color and bold it would not fail because the boldness is not color dependent</q>.

<hr>

In other words, for links not to fail I believe they should meet at least 1 of the following:
- Preserve the default `text-decoration` underline (in the default state, not only on hover/focus):<br>
  <img src="https://user-images.githubusercontent.com/26493779/153768197-bd7aa6e6-f58b-4f3b-a6fe-568ac3dd5c2e.png">
- Use `font-weight: bolder`:<br>
  <img src="https://user-images.githubusercontent.com/26493779/153768225-0042396f-265a-4daf-be90-9024f2b2e410.png">
- Have a contrast ratio of 3:1 with the surrounding text

---
> **Comment by:** danallsop **on** 2025-08-14T14:40:39Z

I've also had [axe-core alerting me to this](https://dequeuniversity.com/rules/axe/4.1/link-in-text-block). In my case, I just used a shade of blue with more contrast, but I think the underline or weighting could also be acceptable. In my personal opinion, the bold looks the best out of those options, but it might lead to unexpected new text wrapping for people putting longer content in there, e.g:

Normal:
<img width="552" height="40" alt="Image" src="https://github.com/user-attachments/assets/e456ac56-2801-4e1c-a176-937a395bbe27" />

Bolder:
<img width="553" height="54" alt="Image" src="https://github.com/user-attachments/assets/4c6b36c4-4d1c-4a71-9066-75d0788b3301" />
